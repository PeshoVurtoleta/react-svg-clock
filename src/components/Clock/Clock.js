import React, {Component} from "react";

export class Clock extends Component {
    currentTime = new Date();

    state = {
        secondsDeg: 0,
        minutesDeg: 0,
        hoursDeg: 0
    };

    getTime = () => {
        return {
            seconds: this.currentTime.getSeconds(),
            minutes: this.currentTime.getMinutes(),
            hours: this.currentTime.getHours()
        }
    };

    setDegrees = () => {
        const {seconds, minutes, hours} = this.getTime();

        return {
            secondsDeg: seconds * 6,
            minutesDeg: minutes * 6 + (seconds * 6) / 60,
            hoursDeg: (hours * 30) + minutes / 2
        }
    };

    componentDidMount() {
        this.setState(() => ({
            ...this.setDegrees()
        }))
    }

    render() {
        const {render} = this.props;
        const {secondsDeg, minutesDeg, hoursDeg} = this.state;

        return <svg viewBox="0 0 40 40">
            {render(secondsDeg, minutesDeg, hoursDeg)}
        </svg>
    }
}
