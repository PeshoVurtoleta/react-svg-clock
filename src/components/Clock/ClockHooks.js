import React, {useEffect, useState} from "react";

export function Clock({render}) {
    const [secondsDeg, setSecondsDeg] = useState(0);
    const [minutesDeg, setMinutesDeg] = useState(0);
    const [hoursDeg, setHoursDeg] = useState(0);

    useEffect(() => {
        const {seconds, minutes, hours} = setTime();

        setSecondsDeg(seconds * 6);
        setMinutesDeg(minutes * 6);
        setHoursDeg((hours * 30) + minutes / 2);

    }, []);

    return <svg viewBox="0 0 40 40">
        {render(secondsDeg, minutesDeg, hoursDeg)}
    </svg>
}