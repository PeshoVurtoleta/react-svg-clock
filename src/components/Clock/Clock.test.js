import React from 'react';
import {shallow, render} from "enzyme";
import {Clock} from './Clock';

const fn = jest.fn;

describe("<Clock/>", () => {
    const wrapper = shallow(<Clock render={fn}/>);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
});

describe('Clock functionality:', () => {
    it("Returns current hours, minutes, seconds:", () => {
        const wrapper = shallow(<Clock render={fn}/>);

        const now = new Date();
        const instance = wrapper.instance();

        expect(instance.getTime()).toEqual({
            hours: now.getHours(),
            minutes: now.getMinutes(),
            seconds: now.getSeconds()
        })
    })
});