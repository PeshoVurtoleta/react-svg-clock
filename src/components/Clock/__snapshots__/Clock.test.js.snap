// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<svg
  viewBox="0 0 40 40"
>
  <Component />
</svg>
`;
